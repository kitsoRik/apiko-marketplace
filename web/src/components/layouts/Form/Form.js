import React from "react";

import "./Form.scss";

const Form = ({ className = "", children, asForm = false, ...props }) => {
	const asComponent = asForm ? "form" : "div";

	return React.createElement(
		asComponent,
		{ className: `form ${className}`, ...props },
		children
	);
};

export default Form;
