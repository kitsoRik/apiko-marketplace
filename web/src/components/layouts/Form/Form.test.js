import React from "react";
import { create } from "react-test-renderer";
import Form from "./Form";

describe("Form", () => {
	it("should rendered", () => {
		const component = create(<Form></Form>);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with children", () => {
		const component = create(<Form>Children</Form>);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with className", () => {
		const component = create(
			<Form className="my-custom-class-name">Children</Form>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
