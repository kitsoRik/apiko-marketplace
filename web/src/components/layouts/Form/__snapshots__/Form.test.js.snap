// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render with children 1`] = `
<div
  className="form "
>
  Children
</div>
`;

exports[`Form should render with className 1`] = `
<div
  className="form my-custom-class-name"
>
  Children
</div>
`;

exports[`Form should rendered 1`] = `
<div
  className="form "
/>
`;
